end_to_end/end_to_end_suite_test.go

```diff
@@ -2254,6 +2254,20 @@ LANGUAGE plpgsql NO SQL;`)
 			Expect(err).To(HaveOccurred())
 			Expect(string(output)).To(ContainSubstring(fmt.Sprintf("Cannot restore a backup taken on a cluster with 5 segments to a cluster with %d segments unless the --resize-cluster flag is used.", segmentCount)))
 		})
+		It("Will clean up segments helper processes after error during restore", func() {
+			command := exec.Command("tar", "-xzf", "resources/4-segment-db-error.tar.gz", "-C", backupDir)
+			mustRunCommand(command)
+			gprestoreCmd := exec.Command(gprestorePath,
+				"--timestamp", "20240226140135",
+				"--redirect-db", "restoredb",
+				"--backup-dir", path.Join(backupDir, "4-segment-db-error"),
+				"--resize-cluster")
+			output, err := gprestoreCmd.CombinedOutput()
+			Expect(err).To(HaveOccurred())
+			Expect(string(output)).To(ContainSubstring(`Error loading data into table public.test: COPY test, line 8193, column test: "32768": ERROR: value "32768" is out of range for type smallint`))
```

Comment:
If I understand this right, backup contains 4 data files for restored table. Each of them has values which are out of range. This values are different in each file.
As files are restored in parallel, the message can contain different values (not only 32768).
This test case in stable flaky on not patched version on my machine.

Reply:
fixed